.ci/merge-fixes.sh

```diff
@@ -14,23 +16,42 @@ else
     git tag -f test_base
     git commit -q -m "Uncommitted changes" --no-allow-empty -a
     for a in $PRs; do
-        git fetch --unshallow --all > /dev/null 2>&1 && echo "Unshallowed."
-        echo "::group::Merging PR https://github.com/$REPO/pull/$a"
         git tag -f test_head
-        $GH pr checkout -b pr-$a $a
-        git checkout -q test_head
-        if git merge --no-edit --squash -q pr-$a; then
+        # We used to pull the branch and merge it (after unshallowing), but when run on PRs that are
+        # based on older releases, it has the side effect of updating to this release,
+        # which may be confusing.
+        #
+        # Here, we obtain the "git am"-able patch of the PR branch.
+        # This also makes it unnecessary to unshallow the repository.
+        #
+        # Considered alternative: Use https://github.com/$REPO/pull/$a.diff,
+        # which squashes everything into one diff without commit metadata.
+        PATH=build/bin:$PATH build/bin/sage-download-file "https://github.com/$REPO/pull/$a.patch" upstream/$a.patch
+        date -u +"%Y-%m-%dT%H:%M:%SZ" > upstream/$a.date  # Record the date, for future reference
+        LAST_SHA=$(sed -n -E '/^From [0-9a-f]{40}/s/^From ([0-9a-f]{40}).*/\1/p' upstream/$a.patch | tail -n 1)
+        echo "::group::Applying PR https://github.com/$REPO/pull/$a @ https://github.com/$REPO/commit/$LAST_SHA as a patch"
+        export GIT_COMMITTER_NAME="$GIT_AUTHOR_NAME applying https://github.com/$REPO/pull/$a @ https://github.com/$REPO/commit/$LAST_SHA"
```

Comment:
I realized `https://github.com/$REPO/commits/$LAST_SHA` is even better.

Compare:
 - https://github.com/sagemath/sage/commit/4a70c1e535bc621bdbe95d8f531375e5700d7c3e
 - https://github.com/sagemath/sage/commits/4a70c1e535bc621bdbe95d8f531375e5700d7c3e

Reply:
Does that work in logs?

Reply:
I want to be careful there so that when the workflow runs in a fork, these links still correctly point to sagemath/sage

Reply:
In the logs only full URLs are clickable, and there is no translation like here. The point is not that it looks shorter, but that the url `https://github.com/sagemath/sage/commit/4a70c1e535bc621bdbe95d8f531375e5700d7c3e` that you are using now is a little bit less nice than the url `https://github.com/sagemath/sage/commits/4a70c1e535bc621bdbe95d8f531375e5700d7c3e`. From the latter one can click on the individual commit to see the patch but it's nicer as a summary of the PR changes when there's more than one commit.

Reply:
Ah thanks, I missed that from how github formatted your comment https://github.com/sagemath/sage/pull/36686#discussion_r1390519957

Reply:
Done in 9dac7295709.

Reply:
The links still work using `sagemath/sage` as the repo even if the commit doesn't belong to that repo but a fork.